server/drafty/drafty.go

```diff
@@ -231,8 +233,102 @@ func toTree(drafty *document) (*node, error) {
 	return &node{children: children}, nil
 }
 
+// given a grapheme iterator, start and end pos, returns a new grapheme iterator
+// containing a slice of graphemes(start->end) from input interator
+func sliceGraphemeClusters(g *uniseg.Graphemes, start, end int) *uniseg.Graphemes {
+	if g == nil {
```

Comment:
Can it be `nil`? How?

Reply:
Yes, you are right, current code scenarios it is never assigned to nil, but since it is a pointer and somehow got assigned to nil, `g.Next()` would panic.

What I can do is return error if g is `nil`